services:
  postgres:
    image: postgres:16.10-alpine3.22
    restart: always
    volumes:
      - ./postgres_init:/docker-entrypoint-initdb.d:ro
      - ./postgres_data:/var/lib/postgresql/data:rw
    ports:
      - "5434:5432"
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U postgres"]
      interval: 30s
      timeout: 20s
      retries: 3
    environment:
      POSTGRES_USER: "postgres"
      POSTGRES_PASSWORD: "postgres"

  hydra:
    image: ghcr.io/hydradatabase/hydra:16-5a258d26896de0f47b08658dc8fa914ad453eeab
    restart: always
    volumes:
      - ./hydra_init:/docker-entrypoint-initdb.d:ro
      - ./hydra_data:/var/lib/postgresql/data:rw
    ports:
      - "5433:5432"
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U postgres"]
      interval: 30s
      timeout: 20s
      retries: 3
    environment:
      POSTGRES_USER: "postgres"
      POSTGRES_PASSWORD: "postgres"


  notebook:
    image: quay.io/jupyter/base-notebook:2025-09-08
    command: "start-notebook.py --NotebookApp.token=jupyter"
    ports: 
      - "8888:8888"
    volumes:
      - ./notebooks:/home/jovyan/work:rw
      - ./datasets:/datasets:ro
    user: root
    environment:
      POSTGRES_USER: "postgres"
      POSTGRES_PASSWORD: "postgres"
      GRANT_SUDO: yes
